### PR TITLE
Remove complex marshalling support from crossgen2

### PR DESCRIPTION
**Pull request by jkoritzinsky**
We still support blittable marshalling as this mechanism is used to decide whether or not to inline a P/Invoke.

Also fix inlining rules for P/Invokes with pointer parameters in CoreCLR (they should be inlinable).

Contributes to #69758

### CODE REVIEW

File: src/coreclr/tools/aot/ILCompiler.ReadyToRun/IL/Stubs/PInvokeILEmitter.cs [hunk 1]
```diff
@@ -43,20 +43,6 @@ private void EmitPInvokeCall(PInvokeILCodeStreams ilCodeStreams)
 
             MetadataType stubHelpersType = InteropTypes.GetStubHelpers(context);
```
Comment (MichalStrehovsky) [inline src/coreclr/tools/aot/ILCompiler.ReadyToRun/IL/Stubs/PInvokeILEmitter.cs:44]: GetStubHelpers can be deleted

**Comment by kant2002** [comment]
I wonder, does this kind of unification allows use more marshallers from NativeAOT in the crossgen, or that would be not practical?

**Comment by jkoritzinsky** [comment]
That's actually going in the other direction. We're removing the marshaller support in crossgen since it isn't being used or tested any more.

This allows us to simplify the logic since we only use the marshallers (at least the non-blittable ones) in NativeAOT after this change.

**Comment by kant2002** [comment]
I suspect this. Does this mean that DllImport would not work in R2R?

**Comment by jkoritzinsky** [comment]
We already do not pre-generate stubs for DllImport in R2R unless System.Private.CoreLib is in your version bubble (composite mode, large version bubble, or System.Private.CoreLib itself), we leave it to the runtime. Since we've moved the BCL and ASP.NET Core to use source-generated interop, the majority of use cases for the R2R support for generating IL stubs is now non-existent.

**Comment by kant2002** [comment]
Thank you! Was not aware about that. Whole thing make sense now

**Review by MichalStrehovsky — APPROVED** [review]
Didn't look at the dllimport.cpp change.

Also cc @dotnet/crossgen-contrib since I'm not the decision maker for crossgen2.

**Review by davidwrighton — APPROVED** [review]

**Pull request merged**